### DISCUSSION TITLE
Unique values in flags where `repeatable` is `true`

### DISCUSSION THREAD

**Discussion by auryark** [bashly-framework/bashly#452] [Ideas]
I think it would be nice if flags with `repeatable: true` only took unique values. I've been doing this kind of filtering using bash's built-in regex operator, so I did a test by manually editing a generated script and adding it to the `parse_requirements` function of a command:

```bash
if [[ -z ${args['--image']+x} ]]; then
  args['--image']="\"$2\""
else
  if [[ ! "${args['--image']}" =~ $2 ]]; then
    args['--image']="${args[--image]} \"$2\""
  fi
fi
```

Maybe that would be a possible solution.

**Comment by DannyBen** [comment]
It is a good idea but:

1. The regex might also filter out *similar* values.
2. Shouldn't setting `repeatable` to also be unique be something that the developer should choose explicitly? I mean, perhaps there are occasions where non-unique are desired?

**Comment by DannyBen** [comment]
I have implemented #454 and #456 to add support for `unique: true` for flags (only when they have `repeatable` and `arg`) and for args (only when they are `repeatable`).

**Reply by auryark** [reply]
Thanks for implementing my suggestion. I didn't have time to come here and thank you. This feature has been so much useful for me.

**Reply by DannyBen** [reply]
Sure thing, you have good ideas :)
